### PR TITLE
1 test for series.min and min implementation

### PR DESCRIPTION
**Pull request by 1e-to**

### CODE REVIEW

File: hpat/datatypes/hpat_pandas_series_functions.py [hunk 1]
```diff
@@ -356,6 +357,63 @@ def hpat_pandas_series_append_impl(self, to_append):
     return hpat_pandas_series_append_impl
 
 
+@overload_method(SeriesType, 'min')
+def hpat_pandas_series_min(self, axis=None, skipna=True, level=None, numeric_only=None):
```
Comment (densmirn) [inline hpat/datatypes/hpat_pandas_series_functions.py]: Missed **kwargs

File: hpat/datatypes/hpat_pandas_series_functions.py [hunk 2]
```diff
@@ -356,6 +357,63 @@ def hpat_pandas_series_append_impl(self, to_append):
...
+    """
+    Pandas Series method :meth:`pandas.Series.min` implementation.
+
+    .. only:: developer
+
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_attr3
```
Comment (densmirn) [inline hpat/datatypes/hpat_pandas_series_functions.py]: Specified incorrect name of the test.

File: hpat/hiframes/pd_series_ext.py [hunk 3]
```diff
@@ -241,6 +241,7 @@ def __init__(self, dmm, fe_type):
 make_attribute_wrapper(SeriesType, 'name', '_name')
 
 
+from hpat.datatypes.hpat_pandas_series_functions import *
```
Comment (densmirn) [inline hpat/hiframes/pd_series_ext.py]: I think we shouldn't move the import. @shssf please correct me if I'm wrong.

File: hpat/datatypes/hpat_pandas_series_functions.py [hunk 4]
```diff
@@ -356,6 +357,63 @@ def hpat_pandas_series_append_impl(self, to_append):
...
+
+    Parameters
+    -----------
+    axis :  {index (0)}, default: None
```
Comment (densmirn) [inline hpat/datatypes/hpat_pandas_series_functions.py]: Please specify type of the argument.

File: hpat/datatypes/hpat_pandas_series_functions.py [hunk 5]
```diff
@@ -356,6 +357,63 @@ def hpat_pandas_series_append_impl(self, to_append):
...
+
+    if not isinstance(self.dtype, (types.Integer, types.Float)):
+        raise TypingError(
+            '{} Currently function supports only numeric values. Given data type: {}'.format(_func_name, self.dtype))
+
+    if not isinstance(skipna, (types.Omitted, bool)):
+        raise TypingError(
+            '{} The parameter must be a boolean type. Given type skipna: {}'.format(_func_name, type(skipna)))
+
+    if not (isinstance(axis, types.Omitted) or axis is None) \
```
Comment (akharche) [inline hpat/datatypes/hpat_pandas_series_functions.py]: If we don't support "axis" parameter, please, mention it in docstring

File: hpat/datatypes/hpat_pandas_series_functions.py (hunk 2 above)
Comment (1e-to) [reply hpat/datatypes/hpat_pandas_series_functions.py]: Why incorrect? This test alredy existed, and it's chech series.min

File: hpat/datatypes/hpat_pandas_series_functions.py [hunk 6]
```diff
@@ -356,6 +357,63 @@ def hpat_pandas_series_append_impl(self, to_append):
...
+            or not (isinstance(level, types.Omitted) or level is None) \
+            or not (isinstance(numeric_only, types.Omitted) or numeric_only is None):
+        raise TypingError(
+            '{} Unsupported parameters. Given axis: {}, level: {}, numeric_only: {}'.format(_func_name, axis, level,
+                                                                                            numeric_only))
```
Comment (akharche) [inline hpat/datatypes/hpat_pandas_series_functions.py]: Please, delete blank string

File: hpat/tests/test_series.py [hunk 7]
```diff
@@ -902,6 +902,14 @@ def test_impl(S):
         S = pd.Series([np.nan, 2., 3.])
         self.assertEqual(hpat_func(S), test_impl(S))
 
+    def test_series_min(self):
+        def test_impl(S):
+            return S.min()
+        hpat_func = hpat.jit(test_impl)
+
+        S = pd.Series([5, 6, 13])
```
Comment (akharche) [inline hpat/tests/test_series.py]: May be, it's better to check not only Integer values? What if we add some nan elements?

File: hpat/datatypes/hpat_pandas_series_functions.py (hunk 1 above)
Comment (shssf) [reply hpat/datatypes/hpat_pandas_series_functions.py]: we don't need kwargs in this place

File: hpat/hiframes/pd_series_ext.py (hunk 3 above)
Comment (shssf) [reply hpat/hiframes/pd_series_ext.py]: Unfortunately this import is still needed. We will remove it in future. If you have better solution how to make new style functions - please let me know.

File: hpat/tests/test_series.py (hunk 7 above)
Comment (shssf) [reply hpat/tests/test_series.py]: and float, and negatives and `None`

File: hpat/datatypes/hpat_pandas_series_functions.py (hunk 2 above)
Comment (densmirn) [reply hpat/datatypes/hpat_pandas_series_functions.py]: Specified test `test_series_attr3` checks series.size.

File: hpat/datatypes/hpat_pandas_series_functions.py (hunk 1 above)
Comment (densmirn) [reply hpat/datatypes/hpat_pandas_series_functions.py]: In this case do we still need to describe kwargs in docstring? I would say not.

File: hpat/hiframes/pd_series_ext.py (hunk 3 above)
Comment (densmirn) [reply hpat/hiframes/pd_series_ext.py]: I did in this way to overlap Array by Series: [#192](https://github.com/IntelPython/hpat/pull/192/files#diff-f13c21aac60fb8f7ae3c9527239d1e17R991-R999)

File: hpat/datatypes/hpat_pandas_series_functions.py [hunk 8]
```diff
@@ -356,6 +357,63 @@ def hpat_pandas_series_append_impl(self, to_append):
...
+         returns :obj:`pandas.Series` object or :obj:`int` or :obj:`float`
+    """
+
+    _func_name = 'Method min().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError(
+            '{} The object must be a pandas.series. Given self: {}'.format(_func_name, self))
+
+    if not isinstance(self.dtype, (types.Integer, types.Float)):
```
Comment (densmirn) [inline hpat/datatypes/hpat_pandas_series_functions.py]: You could use `types.Number` as shortcut for `(types.Integer, types.Float)` .

**Pull request merged**